src/metabase/cmd/endpoint_dox.clj

```diff
@@ -5,22 +5,120 @@
             [metabase.plugins.classloader :as classloader]
             [metabase.util :as u]))
 
+;;;; API docs intro
+
+(defn- api-docs-intro!
+  "Exists just so we can write the intro in Markdown."
+  []
+  (str (slurp "src/metabase/cmd/resources/api-intro.md") "\n\n"))
+
+;;;; API docs section title
+
+(defn- endpoint-ns-name
+  "Creates a name for endpoints in a namespace, like all the endpoints for Alerts."
+  [endpoint]
+  (->> (:ns endpoint)
+       (ns-name)
+       (name)
+       (#(str/split % #"\."))
+       (last)
+       (str/capitalize)))
+
+(defn- section-title
+  "Creates a section title for a set of endpoints."
+  [ep-title]
+  (str/replace (str "## " ep-title "\n\n") #"-" " "))
+
+;;;; API docs section description
+
+(defn- section-description
+  "If there is a namespace docstring, include the docstring with a paragraph break."
+  [ep-data]
+  (let [desc (u/add-period (:doc (meta (:ns (first ep-data)))))]
+    (if (str/blank? desc) desc
+        (str desc "\n\n"))))
+
+;;;; API docs section table of contents
+
+(defn- anchor-link
+  "Converts an endpoint string to an anchor link, like [GET /api/alert](#get-apialert),
+  for use in section tables of contents."
+  [ep-name]
+  (let [al (-> (str "#" (str/lower-case ep-name))
+               (str/replace #"[/:-_]" "")
+               (str/replace " " "-")
+               (#(str "(" % ")")))]
+    (str "[" ep-name "]" al)))
+
+(defn- toc-links
+  "Creates a list of links to endpoints in the relevant namespace."
+  [endpoint]
+  (->> (:endpoint-str endpoint)
+       (#(str/replace % #"[#+`]" ""))
+       (str/trim)
+       (anchor-link)
+       (#(str "  - " %))))
+
+(defn section-toc
+  "Generates a table of contents for endpoints in a section."
+  [ep-data]
+  (str (str/join "\n" (map toc-links ep-data)) "\n\n"))
+
+;;;; API docs section endpoints
+
+(defn- endpoint-str
+  "Creates a name for an endpoint: VERB /path/to/endpoint. Used to build anchor links in the table of contents."
+  [endpoint]
+  (->> (:doc endpoint)
+       (#(str/split % #"\n"))
+       (first)
+       (str/trim)))
```

Comment:
```suggestion
  (-> (:doc endpoint)
      (str/split #"\n")
      first
      str/trim))
```

If you use `->` it automatically passes the result of each form as the first argument of the next form. So if you use this you don't need to introduce an anonymous function to pass the docstring to `str/split`